File: Lab/Lab2/lab2.java
import java.util.HashMap;
import java.util.Map;

public class lab2{
    static boolean anagram(String s, String t){
        Map <Character, Integer> lettersMap = new HashMap<Character, Integer>();

        s = s.toLowerCase();
        t = t.toLowerCase();
        int sLength = 0;
        int tLength = 0;

        //if there's a space in the string, I want to skip and add to actual string length
        for (int i = 0; i < s.length(); i++){
            if(s.charAt(i) == ' '){
                continue;
            }
            else{
                sLength++;
            }
        }

        for (int i = 0; i < t.length(); i++){
            if(t.charAt(i) == ' '){
                continue;
            }
            else{
                tLength++;
            }
        }
        
        //compare the string lengths without spaces
        if (sLength != tLength){
            return false;
        }

        for (int i = 0; i < s.length(); i++){
            if(!lettersMap.containsKey(s.charAt(i))){
                if (s.charAt(i) == ' '){
                    continue;
                }
                else{
                    lettersMap.put(s.charAt(i), 1);
                }
            }
            else{
                int count = lettersMap.get(s.charAt(i)) + 1;
                lettersMap.put(s.charAt(i), count);
            }
        }

        for (int i = 0; i < t.length(); i++){
            if (lettersMap.containsKey(t.charAt(i))){
                int minus = lettersMap.get(t.charAt(i)) - 1;
                lettersMap.put(t.charAt(i), minus);
            }
            if(lettersMap.get(t.charAt(i)) == null){
                continue;
            }
            if (lettersMap.get(t.charAt(i)) == -1 ){
                return false;
            }
        }

        for (int i = 0; i < sLength; i++){
            if(lettersMap.get(s.charAt(i)) == null){
                continue;
            }
            if(lettersMap.get(s.charAt(i)) != 0){
                return false;
            }
        }
        return true;
    }
    
    public static void main(String[] args) {
        System.out.println(anagram(" Dormitory", "Dirty Room"));
    }
}

File: Lab/Lab2/lab2.py
def anagram(s, t):
    lettersMap = {}

    s = s.lower()
    t = t.lower()
    sRange = len(s)
    tRange = len(t)
    sLength = 0
    tLength = 0

    #checking lengths
    for i in range(sRange):
        if s[i] == ' ':
            continue
        else:
            sLength += 1

    for i in range(tRange):
        if t[i] == ' ':
            continue
        else:
            tLength += 1

    if sLength != tLength:
        return False
    
    #input into dictionary if there isn't a key already
    for i in range(sRange):
        if not s[i] in lettersMap:
            if s[i] == ' ':
                continue
            else: 
                lettersMap.update({s[i]: 1})
        else:
            count = lettersMap.get(s[i]) + 1
            lettersMap.update({s[i]: count})
        
    #decrement key value if the same letter is found in string t
    for i in range(tRange):
        if t[i] in lettersMap:
            minus = lettersMap.get(t[i]) - 1
            lettersMap.update({t[i]: minus})
        if lettersMap.get(t[i]) == None:
            continue
        if lettersMap.get(t[i]) == -1:
            return False

    #condition to ensure that all key values = 0
    for i in range(sRange):
        if lettersMap.get(s[i]) == None:
            continue
        if lettersMap.get(s[i]) != 0:
            return False
        
    return True

print(anagram(" Dormitory", "Dirty Room"))

